CONTRIBUTING.md

```diff
@@ -252,6 +252,74 @@ There may, at times, need to be exceptions where breaking changes are allowed in
 discretion of the project's maintainers, and must be carefully considered before merging. An example of an allowed
 breaking change might be a fix for a behavioral bug that was released in an initial minor version (such as `v0.3.0`).
 
+
+## API conventions
+
+This project follows the [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required). Minor modifications or additions to the conventions are listed below.
+
+### Optional vs. Required
+
+* Status fields MUST be optional. Our controllers are patching selected fields instead of updating the entire status in every reconciliation.
+
+Optional fields have the following properties:
+* An optional field MUST be marked with `+optional` and include an `omitempty` JSON tag.
+* Fields SHOULD be pointers if the nil and the zero values (by Go standards) have semantic differences.
+  * Note: This doesn't apply to map or slice types as they are assignable to `nil`.
```

Comment:
should we just avoid maps in general? https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps

Reply:
In general yes, but there are exceptions like `The only exceptions are pure maps in the API (currently, labels, selectors, annotations, data)`